### PR TITLE
Add "Buy BSQ" button next to trade fee selector

### LINKED ISSUE #4579: Trade Step 1: Add "click here to buy BSQ" link next to trade fee selector
In step 1 of a trade, both the maker and the taker have to choose how they pay for the trade fee: in BTC or BSQ.

Most people don't know what BSQ is or where they can get it, especially new users. They probably assume that, since they don't already have any BSQ, they can't choose that option.

Therefore, Bisq should add a link to the BSQ market next to the BTC/BSQ fee selector, like this:

Screenshot 2020-09-30 at 09 48 11

Clicking that link should open the "SELL BTC" view, with the BSQ currency pre-selected.

### PR DESCRIPTION
**Pull request by Jakub-CZ**
Fixes #4579 

New button is added in the _CREATE OFFER_ and _TAKE OFFER_ tabs. When clicked, it shows a popup and then brings the user to the Sell Offer Book (_SELL BITCOIN_), with BSQ preselected in the currency filter.

The button is visible only when the fee currency toggles are visible (relevant in the _CREATE OFFER_ tab).

## Screenshots

start creating or take any offer ->

Buy BSQ 1

-> click "BUY BSQ" ->

Buy BSQ popup

-> click "BUY BSQ" in popup ->

image

## Situations when the button is clicked (unchecked are still buggy):

- [x] user is creating/taking offer to _BUY BTC_, and under _SELL BTC_ the tab "_CREATE OFFER_" is **not** active (it may be open but not selected);
- [x] user is creating/taking offer to _BUY BTC_, and under _SELL BTC_ the active tab is "_CREATE OFFER_";
- [x] user is creating offer to _SELL BTC_;
- [x] user is taking offer to _SELL BTC_.

## TODOs from discussion below:
- [x] close the current tab instead of just navigating away from it to the Offer Book;
- [x] don't display the button when the BSQ balance is already high enough.

## Known bug:

- Certain combination of the source and destination views may cause the "_Filter by currency_" combo box to receive focus, clearing it as a result. At that point, the value "BSQ" is still in effect, but any user action after that clears the filter. (More on that in the comments below.)

## Discussion:

1. If _CREATE OFFER_ to _SELL BTC_ is already open, user will be unable to create an offer, as the button _CREATE NEW OFFER TO BUY BSQ_ will be disabled. There is no indication why. A tooltip explaining why it's disabled may be helpful.
2. Should the popup include the "_don't remind me again_" option?
3. Should we hide the button when we're creating/taking an offer to trade BTC for BSQ already?
4. Having to create an altcoin account for BSQ seems like an unnecessary step. Can we have a built-in BSQ account, ideally one that will update the address to the next unused one dynamically?

### CODE REVIEW

**Comment by Conza88** [comment]
Can we please add link to wiki on bsq etc? 
So folks can easily see trading value of purchasing etc

**Comment by Jakub-CZ** [comment]
I've been trying to resolve the combo box focus issue for a long time, with not much success. I tried the following:
- disable the combo box temporarily,
- disable the focus listener of the combo box temporarily, or `setFocusTraversable(false)`,
- `requestFocus()` on another element, right around the time the tab switching takes place (this resolved only some cases, mostly the 2nd item in my checkbox list above),
- and much more.

These either had no effect, or caused the combo box not to reflect `sellScreenCurrencyCode` in Preferences.

Also, when the combo box receives focus, it's probably happening too early, as indicated by the fact that the drop-down list is rendered incorrectly - it's shifted to the left:

Buy BSQ tab select - combobox focus bug

### Some options that I see:
- carefuly rework AutocompleteComboBox. Instead of relying on onFocus, use onClick; I tried that already, but then the value in the combo box wasn't in line with `sellScreenCurrencyCode` in Preferences.
- instead of clearing the value when focused, just select the contents of the edit text field; thus, the previous value will not be lost, and users can still very easily filter by typing right away.

**Comment by Jakub-CZ** [comment]
> Can we please add link to wiki on bsq etc?
> So folks can easily see trading value of purchasing etc

This one? https://bisq.wiki/Trading_fees

Is this not enough?
image

**Comment by Conza88** [comment]
Yeah that one. Just in the pop-up I think. Guides folks to wiki if they want to know benefits etc.

I think ideally should have bsq amount somewhere more prominent.
________________________________

**Comment by pazza83** [comment]
@Jakub-CZ looks great.

I had a read but not sure what happens when they click the 'Buy BSQ' button?

Are they taken to the Buy > BSQ/BTC market?

**Comment by Jakub-CZ** [comment]
Yep, except it's **Sell**, as in "selling BTC for BSQ".

I've added the final screenshot in the description at the top.

**Comment by pazza83** [comment]
Thanks, that is great. Hopefully will get more new users to buy some BSQ :)

File: desktop/src/main/java/bisq/desktop/main/offer/OfferViewUtil.java [hunk 1]
```diff
@@ -33,4 +47,26 @@ public static Label createPopOverLabel(String text) {
...
+        buyBsqButton.setOnAction(e -> new Popup().headLine(buyBsqText)
+                .information(Res.get("createOffer.buyBsq.popupMessage"))
+                .actionButtonText(buyBsqText)
+                .buttonAlignment(HPos.CENTER)
+                .onAction(() -> {
+                    preferences.setSellScreenCurrencyCode("BSQ");
+                    navigation.navigateTo(MainView.class, SellOfferView.class, OfferBookView.class);
+                }).show());
+
+        final VBox buyBsqButtonVBox = new VBox(buyBsqButton);
```
Comment (chimp1984) [inline desktop/src/main/java/bisq/desktop/main/offer/OfferViewUtil.java]: Why a VBox is created if there is only one element in it?
The `final` in local variables should not be used (I know much of old code have it as well, but current code style suggest to not use it as unnecessary verbose).

**Comment by chimp1984** [comment]
I had a quick look, but did not see anything obvious which could cause the problems you described. Would need to debug locally but too busy atm...

**Comment by Conza88** [comment]
I really think these concerns need to be addressed before sending shiteload more people to buy bsq: https://github.com/bisq-network/bisq/issues/5046

**Comment by sqrrm** [comment]
For the case where someone is selling BTC I think the current sale tab needs to be closed and a new one opened for buying BSQ since there can only be one BTC sell tab open. That could perhaps be a bit confusing but I think allowing for multiple create offer tabs would be more trouble. Perhaps include a warning that the current BTC sell tab will be closed in the popup as well for that case.

Auto creating a BSQ account is reasonable and something I did for the not yet completed atomic BSQ trades. It's a bit more problematic for the current model as the receive address needs to be available throughout the trade process and our current model expects 1 single address per account. I think we could create a BSQ account with the first unused BSQ address if there is no other BSQ account. That would make it somewhat easier to get started with BSQ in the current state.

@Jakub-CZ I can't give any input on the UI elements not aligning and the focus, @ripcurlx knows that best I think.

File: desktop/src/main/java/bisq/desktop/main/offer/OfferViewUtil.java [hunk 2]
```diff
@@ -33,4 +47,26 @@ public static Label createPopOverLabel(String text) {
         label.setPadding(new Insets(10));
         return label;
     }
+
+    public static Tuple2<AutoTooltipButton, VBox> createBuyBsqButtonBox(Navigation navigation, Preferences preferences) {
+        String buyBsqText = Res.get("shared.buyCurrency", "BSQ");
+        var buyBsqButton = new AutoTooltipButton(buyBsqText);
+        buyBsqButton.getStyleClass().add("action-button");
+        buyBsqButton.setStyle("-fx-pref-height: 20; -fx-padding: 3 8 3 8; -fx-font-size: 0.769em; -fx-border-radius: 5;");
```
Comment (ripcurlx) [inline desktop/src/main/java/bisq/desktop/main/offer/OfferViewUtil.java]: Please move this into the css file and use another class instead.

**Review by ripcurlx — COMMENTED** [review]
I think having on the small green button `Buy BSQ` might confuse noobs what will happen if they click on this.
Maybe it should be something more like `Get BSQ` or `How to get BSQ` 👈  @m52go ?
We could use this popup afterwards not only to push the user to buy BSQ, but also show the possibility to earn BSQ by contributing.

Maybe we could also only show the Buy BSQ button if the trade has insufficient funds to pay the trading fee with BSQ.

Regarding the combobox question. So the goal is to open the offer book to sell BTC and have all BSQ offers selected? Atm it switches to the offer book setting the focus in the currency combobox.


> For the case where someone is selling BTC I think the current sale tab needs to be closed and a new one opened for buying BSQ since there can only be one BTC sell tab open. That could perhaps be a bit confusing but I think allowing for multiple create offer tabs would be more trouble. Perhaps include a warning that the current BTC sell tab will be closed in the popup as well for that case.

That is the current behavior. It will just replace the current `Take offer` tab that was used for selling BTC in the first place. Maybe we just add a `don't show again` popup if a user is trying to take different offers to reduce confusion where the old `take offer` tab vanished to.

**Comment by Jakub-CZ** [comment]
@sqrrm
I thought about closing the tab as well, but it might be annoying to the user if they already filled in some of the fields (the button doesn't appear until at least the volume is filled in). Adding 

Also, closing the tab may get around the UI issue I'm having right now.

---
@ripcurlx 

> Regarding the combobox question. So the goal is to open the offer book to sell BTC and have all BSQ offers selected? Atm it switches to the offer book setting the focus in the currency combobox.

Yes, that is the goal. The fact that it focuses on the currency combobox is an unintended side effect of the current design. Looks like the Offer Book view gets invalidated as a result of switching from Take Offer. I don't know why the same doesn't happed when switching from BUY BTC.

> only show the Buy BSQ button if the trade has insufficient funds

Good idea. Will look into that.

**Review by wallclockbuilder — DISMISSED** [review]
UtAck

**Comment by Conza88** [comment]
> I think having on the small green button `Buy BSQ` might confuse noobs what will happen if they click on this.
> Maybe it should be something more like `Get BSQ` or `How to get BSQ` 👈 @m52go ?
> We could use this popup afterwards not only to push the user to buy BSQ, but also show the possibility to earn BSQ by contributing.

I think it may certainly confuse any noobs, and even old crew. Also it's like a check-out, instead of one-click amazon - you're putting up barriers for people to think about before making the trade. All else being equal, they'll likely be less inclined to go through with it. Less trades, not good for Bisq. 

If the UX process was smooth for buying BSQ, I'd be less hesitant. Definitely not when it's a battle! 

Shifting to a **(i) How to get BSQ** would not be a barrier to the final trade, but an FYI for future purposes.

**Comment by cd2357** [comment]
@Jakub-CZ any updates or roadblocks? Been looking forward to this PR for a long time :)

**Comment by Jakub-CZ** [comment]
@cd2357 I'll try to get back to this during the upcoming weekend.

File: desktop/src/main/java/bisq/desktop/main/offer/OfferViewUtil.java (hunk 2 above)
Comment (Jakub-CZ) [reply desktop/src/main/java/bisq/desktop/main/offer/OfferViewUtil.java]: done

**Comment by Jakub-CZ** [comment]
Updates since last review:
- added link to relevant wiki page;
- button styling moved to a CSS file;
- button only displayed when not enough BSQ funds to pay for the fee.

There are still buggy scenarios where the "Filter by currency" combo box gets cleared intentionally, as a result of receiving focus. Switching to BUY BTC and then back to SELL BTC activates the desired filter (BSQ).

This needs more debugging and fixing.

**Comment by cd2357** [comment]
Tried it out, looks promising.

You're right, it can get tricky with the closing of tabs, updating states of UI components, etc.

To keep things simple, it might be enough to not do any UI transitions and navigation to the "Sell BTC / Buy BSQ" view, but instead to just open the "Buy BSQ" popup. If the user is interested, he will follow the links and read into it. The popup's button could then link to [the step-by-step guide on how to go about and buy some BSQ](https://bisq.wiki/Paying_trading_fees_with_BSQ#Acquire_BSQ).

**Comment by ripcurlx** [comment]
@Jakub-CZ Are you still on this PR?

**Comment by Jakub-CZ** [comment]
> @Jakub-CZ Are you still on this PR?

Sorry, I didn't have time to get back to it. The UI bugs are difficult to iron out.

If anyone wants to take it over and finish it, I wouldn't mind.

**Comment by ripcurlx** [comment]
> > @Jakub-CZ Are you still on this PR?
> 
> Sorry, I didn't have time to get back to it. The UI bugs are difficult to iron out.
> 
> If anyone wants to take it over and finish it, I wouldn't mind.

I'll give it a shot if I can iron out the remaining issues of this PR.

**Comment by ripcurlx** [comment]
Just re-based on top of current master.

**Comment by ripcurlx** [comment]
Although this is kind of repetitive, I see following open points/issues:
- [ ] 1. If `Create offer` is already open it will focus in the `Sell BTC` tab the Filter by Currency dropdown, instead of pre-selecting `BSQ`
Bildschirmfoto 2021-09-30 um 11 28 50

- [x] 2. Just showing `Buy BSQ` IMO is keeping the user in the void, what will happen when clicking on it.
Bildschirmfoto 2021-09-30 um 11 31 51
I think it needs at least an additional help link as suggested in one of @pedromvpg 's screens
Bildschirmfoto 2021-09-30 um 11 33 01

- [x] 3. Having `Create Offer` open in `Sell BTC` section, will close the `Create Offer` Tab when clicking on `Buy BSQ` in the `Buy BTC > Take offer` or `Sell BTC > Create Offer` section.
**Before**
Bildschirmfoto 2021-09-30 um 11 49 22

**After clicking on `Buy BSQ`**
Bildschirmfoto 2021-09-30 um 11 40 51

- [x]  4. Don't show `Buy BSQ` when trying to buy BSQ
Bildschirmfoto 2021-09-30 um 11 44 34

- [x] 5. Process of creating a BSQ account is unclear for new users
When trying to buy BSQ user is guided to create a matching payment account.
Bildschirmfoto 2021-09-30 um 11 54 29
Clicking on `Go to "Account"` will open the account tab in the wrong section and if the user manages to understand that she needs to go to `Altcoin Accounts` and search for BSQ. Where do I get the BSQ address from?
Bildschirmfoto 2021-09-30 um 11 57 27
I do think as suggested by @Jakub-CZ that we should automatically create a BSQ Altcoin account and provide the user with the information where she can find the BSQ wallet later on for confirming a trade. 
We should also add in the `Confirm Payment Receipt` screen similar to the `Start payment` screen a button link to the BSQ wallet. In this case maybe something like `Show Payment in BSQ wallet`.

@Jakub-CZ Did I miss any open points?
@Conza88 @pazza83 Any other open points from your point of view that need to be addressed in this PR?

**Comment by ripcurlx** [comment]
@5. Something like this for confirming the payment received.
Bildschirmfoto 2021-09-30 um 12 22 11
Bildschirmfoto 2021-09-30 um 12 22 19

@m52go Any ideas for the perfect text?

I wouldn't confirm automatically right now, so users are aware of the BSQ wallet and it is less a risky addition.

File: core/src/main/resources/i18n/displayStrings.properties [hunk 3]
```diff
@@ -480,6 +480,11 @@ createOffer.triggerPrice.tooltip=As protection against drastic price movements y
 createOffer.triggerPrice.invalid.tooLow=Value must be higher than {0}
 createOffer.triggerPrice.invalid.tooHigh=Value must be lower than {0}
 
+createOffer.buyBsq.popupMessage=Trading fees are paid to fund the Bisq DAO when making or taking an offer. Fees can be paid in BSQ or BTC.\n\n\
+  Bisq encourages traders to use BSQ by offering a discount of approximately 50%, since BSQ fees help fund Bisq's development. \
+  This discount varies as the BSQ/BTC rate fluctuates. To maintain the 50% discount target, trading fees are updated every cycle as necessary.\n\n\
+  For more details see [HYPERLINK:https://bisq.wiki/Trading_fees]
```
Comment (m52go) [inline core/src/main/resources/i18n/displayStrings.properties]:
```suggestion
createOffer.buyBsq.popupMessage=Trading fees are paid to fund the Bisq DAO. Fees can be paid in BSQ or BTC.\n\n\
  BSQ fees directly help fund Bisq's development, so Bisq encourages traders to use BSQ by offering a 50% discount on trading fees. \
  This discount varies as the BSQ/BTC rate fluctuates. To maintain the 50% discount target, trading fees are updated every cycle as necessary.\n\n\
  For more about fees, see [HYPERLINK:https://bisq.wiki/Trading_fees]. For more about the Bisq DAO, see [HYPERLINK:https://bisq.wiki/Introduction_to_the_DAO#The_Bisq_DAO].
```

Just made some wording changes and added a link to the DAO introduction on the wiki.

**Comment by m52go** [comment]
I recommend adding more context to this area in addition to the "Learn more" link. Some people still perceive BSQ to be a shitcoin, so having something short like...

"BSQ is colored BTC that helps fund Bisq developers. <a href="#">Learn More</a>" 

...next to the `BUY BSQ` button could help a lot.

Screenshot from 2021-10-01 13-01-42

**Comment by Conza88** [comment]
Fark yes to the learn more and context setting gents 👏👌🎯. 

Also very good point about coloured Bitcoin. 
Hence why I've commented elsewhere, we need BSQ to be separate compared to the shitcoins. We literally put it in that category (oof). 

But yeah, great first step 👍.

**Comment by pazza83** [comment]
> @Conza88 @pazza83 Any other open points from your point of view that need to be addressed in this PR?

@ripcurlx thanks for this and the useful screen shots showing the flow.

> 1. If Create offer is already open it will focus in the Sell BTC tab the Filter by Currency dropdown, instead of pre-selecting BSQ

That seems like it would be a little confusing. Any way to get the user straight to the Sell BTC / Buy BSQ screen?

> 2. Just showing Buy BSQ IMO is keeping the user in the void, what will happen when clicking on it.

I Agree, @m52go's solution https://github.com/bisq-network/bisq/pull/5055#issuecomment-932399997 looks good 

> 3. Having Create Offer open in Sell BTC section, will close the Create Offer Tab when clicking on Buy BSQ in the Buy BTC > Take offer or Sell BTC > Create Offer section.

I think this seems ok for it to close.

4. Don't show Buy BSQ when trying to buy BSQ

I think the user should be pushed towards the take offer page, not the make offer page, (more immediate)

> 5. Process of creating a BSQ account is unclear for new users. When trying to buy BSQ user is guided to create a matching payment account.

Bisq handles this well with pop up to create an account.

> Clicking on Go to "Account" will open the account tab in the wrong section and if the user manages to understand that she needs to go to Altcoin Accounts and search for BSQ. Where do I get the BSQ address from?

Maybe pop up text can be updated to include a link to the wiki and mention it is an new altcoin account. @Conza88 previously felt BSQ should have it's own tab. I think this is good but one step at a time :)

> I do think as suggested by @Jakub-CZ that we should automatically create a BSQ Altcoin account and provide the user with the information where she can find the BSQ wallet later on for confirming a trade.

Agreed

> We should also add in the Confirm Payment Receipt screen similar to the Start payment screen a button link to the BSQ wallet. In this case maybe something like Show Payment in BSQ wallet.

Agreed

**Comment by ripcurlx** [comment]
> That seems like it would be a little confusing. Any way to get the user straight to the Sell BTC / Buy BSQ screen?

That is what it should do - I just pointed out the UI error with the Combox of the current implementation.

**Comment by ripcurlx** [comment]
@Jakub-CZ Was there a reason why you wanted to close the `Create Offer` view when switching the user to the Sell Bitcoin (Buy BSQ) screen?

**Comment by ripcurlx** [comment]
Now with more context. Clicking on `Learn More` opens the popup, clicking on `Buy BSQ` brings you to the Buy BSQ offer book.
Bildschirmfoto 2021-10-06 um 12 13 35
Bildschirmfoto 2021-10-06 um 12 13 40
Bildschirmfoto 2021-10-06 um 12 14 08

**Comment by Jakub-CZ** [comment]
@ripcurlx:

I think there's no reason to close `Create Offer (to buy BTC)`.

The suggestion was to close `Create Offer (to sell BTC)`. There were two reasons:
1. If we leave it open then the user won't be able to create a new offer to buy BSQ. (also noted by @sqrrm)
2. I was hoping that closing the tab programmatically would get around the UI bug (unwanted focus on the currency combo box).

**Comment by pazza83** [comment]
Hi @ripcurlx 

Should this pop up also include a link to: https://bisq.wiki/Trading_BSQ

136184219-96b9efe1-92fc-4437-9667-cbd1570bfa43

**Comment by Conza88** [comment]
> @Conza88 previously felt BSQ should have it's own tab. I think this is good but one step at a time :)

Yep ha. All in bolded below. 

_"8. Improve BSQ overall

Implement Segwit for BSQ
**[Improve BSQ account creation & purchase process](https://github.com/bisq-network/bisq/issues/5046)**
Improve BSQ accessibility through OTC market and bonded reputation
Add Low priority payment type to BSQ
BSQ balance on wallet info screen includes merit BSQ
BSQ Fees equivalent not correctly calulated into BTC in info tooltip
Very poor UX with buying BSQ. BSQ should not be in the 'alt-coin' camp either imo. Coloured BTC is better. Make more present/clear in app (without implying its better than BTC)."_

**Comment by chimp1984** [comment]
FYI @sqrrm 's atomic BSQ-BTC swap will be ready soon...

**Comment by ripcurlx** [comment]
> FYI @sqrrm 's atomic BSQ-BTC swap will be ready soon...

Even better :smile: That will speed-up the BSQ purchase process even more.

**Comment by ripcurlx** [comment]
Finally I found time to continue on this PR. Unfortunately it isn't possible to set a tooltip for a disabled node (mouse events are disabled), so I needed to wrap it into a StackPane putting a label on top to mimick this IMO standard behavior. TODO: Move into a re-usable component.

Bildschirmfoto 2021-10-11 um 11 19 53

**Comment by ripcurlx** [comment]
The current process is now as follows for no-BSQ-coiner.
Someone wants to buy BTC via Revolut
Bildschirmfoto 2021-10-13 um 11 24 31
She decides she want to save money and support the project and clicks on Buy BSQ
Bildschirmfoto 2021-10-13 um 11 24 37
She clicks on `Take offer to buy BSQ`
Bildschirmfoto 2021-10-13 um 11 30 08
She clicks on `Create Account` (Account is created and the offer is taken and opens the take offer screen). Probably it should be named `Create Account and Take Offer`. Also there shouldn't be a `Buy BSQ` hint when the user tries to buy BSQ.
She proceeds with purchasing BSQ
Bildschirmfoto 2021-10-13 um 11 30 23
When needing to confirm the BSQ transfer there is now the `Show Payment in BSQ Wallet` Button to open the BSQ wallet. Further optimization is not necessary as BSQ <>BTC atomic swap offers will be available soon.
Bildschirmfoto 2021-10-13 um 11 32 06
Bildschirmfoto 2021-10-13 um 11 32 36
After completing the trade she can select BSQ in the offer creation process. (ATM the BSQ Button is not gone as it is not updated live)
Bildschirmfoto 2021-10-13 um 11 32 51

@pazza83 @Conza88 

so IMO following points are still open to make this PR mergable:

- [x] Don't show `Buy BSQ` hint when trying to buy BSQ.
- [x] BSQ Button has to be updated live (on view activation) so it is not displayed anymore after successful BSQ purchase
- [ ] The Combobox UI glitch mentioned in https://github.com/bisq-network/bisq/pull/5055#issuecomment-931183094
- [x] Add informational popup if an BSQ instant account is created

**Comment by ripcurlx** [comment]
I fixed the remaining issues except the Combobox UI glitch which I'll attend soon. @m52go Would be great to get also feedback on the wording of the newly introduced strings. Thanks!

**Comment by ripcurlx** [comment]
- [x] Add popup after account creation to improve UX for new users

**Comment by ripcurlx** [comment]
@m52go Could you please have a look at the strings used in this popup? Thanks!

Bildschirmfoto 2021-10-15 um 11 05 03

**Comment by m52go** [comment]
I was hoping to review the strings in context on regtest but don't see the new `Buy BSQ` button when making/taking offers to buy/sell. 

Does it only show in certain special conditions?

Screenshot from 2021-10-15 10-08-20

Screenshot from 2021-10-15 10-08-42

**Comment by ripcurlx** [comment]
> I was hoping to review the strings in context on regtest but don't see the new `Buy BSQ` button when making/taking offers to buy/sell.
> 
> Does it only show in certain special conditions?
> 
> Screenshot from 2021-10-15 10-08-20
> 
> Screenshot from 2021-10-15 10-08-42

Yes, it is only displayed if you have too little BSQ to pay the trading fees. So best way is to create a new user on Regtest.

File: core/src/main/resources/i18n/displayStrings.properties [hunk 4]
```diff
@@ -439,6 +442,13 @@ offerbook.info.buyAtFixedPrice=You will buy at this fixed price.
 offerbook.info.sellAtFixedPrice=You will sell at this fixed price.
 offerbook.info.noArbitrationInUserLanguage=In case of a dispute, please note that arbitration for this offer will be handled in {0}. Language is currently set to {1}.
 offerbook.info.roundedFiatVolume=The amount was rounded to increase the privacy of your trade.
+offerbook.info.accountCreated.headline=Congratulations
+offerbook.info.accountCreated.message=You''ve just successfully created a BSQ payment account.\n\
+  Your account can be found under Account > Altcoins Accounts > {0} and your BSQ wallet under DAO > BSQ Wallet.\n\n
```
Comment (m52go) [inline core/src/main/resources/i18n/displayStrings.properties:448]:
```suggestion
offerbook.info.accountCreated.message=You''ve just successfully created a BSQ payment account.\n\n\
  See details of this payment account at Account > Altcoins Accounts > {0}. See details of your BSQ wallet at DAO > BSQ Wallet.\n\n
```

File: core/src/main/resources/i18n/displayStrings.properties [hunk 5]
```diff
@@ -439,6 +442,13 @@ offerbook.info.buyAtFixedPrice=You will buy at this fixed price.
...
+offerbook.info.accountCreated.tradeInstant=As you've chosen to take a BSQ instant offer a BSQ instant account was created for you. \
+  For instant trading it is required that both trading peers are online to be able \
+  to complete the trade in less than 1 hour.\n\n
+offerbook.info.accountCreated.takeOffer=You can no proceed to take this offer after closing this popup.
```
Comment (m52go) [inline core/src/main/resources/i18n/displayStrings.properties]:
```suggestion
offerbook.info.accountCreated.takeOffer=You can now proceed to take this offer after closing this popup.
```

File: core/src/main/resources/i18n/displayStrings.properties [hunk 6]
```diff
@@ -439,6 +442,13 @@ offerbook.info.buyAtFixedPrice=You will buy at this fixed price.
 offerbook.info.sellAtFixedPrice=You will sell at this fixed price.
 offerbook.info.noArbitrationInUserLanguage=In case of a dispute, please note that arbitration for this offer will be handled in {0}. Language is currently set to {1}.
 offerbook.info.roundedFiatVolume=The amount was rounded to increase the privacy of your trade.
+offerbook.info.accountCreated.headline=Congratulations
+offerbook.info.accountCreated.message=You''ve just successfully created a BSQ payment account.\n\
+  Your account can be found under Account > Altcoins Accounts > {0} and your BSQ wallet under DAO > BSQ Wallet.\n\n
+offerbook.info.accountCreated.tradeInstant=As you've chosen to take a BSQ instant offer a BSQ instant account was created for you. \
+  For instant trading it is required that both trading peers are online to be able \
+  to complete the trade in less than 1 hour.\n\n
```
Comment (m52go) [inline core/src/main/resources/i18n/displayStrings.properties]:
```suggestion
offerbook.info.accountCreated.tradeInstant=You've chosen to take a BSQ instant offer, so a BSQ instant payment account was created for you. \
  Be aware that instant trades are meant to be completed within 1 hour, \
  so you should be online and available for the next 1 hour.\n\n
```

**Review by chimp1984 — DISMISSED** [review]
utACK

**Comment by chimp1984** [comment]
I think the "BSQ is colored coin..." can be removed then the truncating with smaller width is not happening.

Screenshot 2021-11-05 at 01 36 06

Those methods are a bit weird as they mix up buyer/seller sides (isBuyBsqOffer / !isBuyOffer):
```
public boolean isBuyBsqOffer() {
        return !isBuyOffer() && getOffer().getCurrencyCode().equals("BSQ");
    }
```

File: desktop/src/main/java/bisq/desktop/main/offer/OfferViewUtil.java (hunk 1 above)
Comment (ripcurlx) [reply desktop/src/main/java/bisq/desktop/main/offer/OfferViewUtil.java]: Good question - I inherited this part, but didn't looked in the exact necessity of this layout. I'll simplify this positioning.

**Comment by ripcurlx** [comment]
> public boolean isBuyBsqOffer() {
>         return !isBuyOffer() && getOffer().getCurrencyCode().equals("BSQ");
>     }

Yes, this is a little bit confusing because of the fact that when you buy an altcoin you actually selling Bitcoin. I can add an additional comment to make this less confusing. But actually the naming of `isBuyBsqOffer` is accurate.

**Comment by ripcurlx** [comment]
> I think the "BSQ is colored coin..." can be removed then the truncating with smaller width is not happening.
> 
> Screenshot 2021-11-05 at 01 36 06

I'll adapt the scaling behavior so that the informational box will truncate first and within the informational box the longer informational text.

**Comment by chimp1984** [comment]
Yes the buy/sell direction is always referred to BTC. Only in the UI we swap it for altcoins as its not intuitive otherwise.
Maybe use a different term instead of buy/sell?

**Comment by ripcurlx** [comment]
> > I think the "BSQ is colored coin..." can be removed then the truncating with smaller width is not happening.
> > Screenshot 2021-11-05 at 01 36 06
> 
> I'll adapt the scaling behavior so that the informational box will truncate first and within the informational box the longer informational text.

I've changed the behavior so it is not truncating at 1280:
Bildschirmfoto 2021-11-08 um 10 53 55
and will look as follows with the smallest width possible:
Bildschirmfoto 2021-11-08 um 10 54 01

**Review by chimp1984 — APPROVED** [review]
utACK

**Review by ripcurlx — APPROVED** [review]
ACK - based on https://github.com/bisq-network/bisq/pull/5055#pullrequestreview-799915337

**Pull request merged**